# Práctica Clase 2

1) Abrir el archivo "Emisiones_CO2.csv" y cargar sus datos en un diccionario.

In [1]:
import os
archivo = open('Emisiones_CO2.csv', 'r')

In [89]:
dicc_emisiones = {  'cod_pais' : [],
                    'nom_pais' : [],
                    'region' : [],
                    'anio' : [],
                    'co2' : [],
                    'co2_percapita' : []}

In [19]:

def conv(x):

    x1=x.replace('.','')
    x2=x1.replace(',','.')

    return x2

def arrco2(x):

    if x == '':
        x2='0'
    else:
        x2=conv(x)
    return x2

##El original

archivo = open('Emisiones_CO2.csv', 'r')

la=archivo.readlines()
la.pop(0)
#print(la)

l_cod=[]
l_nom=[]
l_reg=[]
l_ano=[]
l_co2=[]
l_ccap=[]

for i in la:

    info=i.split('|')

    l_cod.append(info[0])  ##El original
    l_nom.append(info[1])
    l_reg.append(info[2])
    l_ano.append(int(info[3]))
    l_co2.append(float(arrco2(info[4])))
    l_ccap.append(info[5])

archivo.close()

k=list(dicc_emisiones.keys())

dicc_emisiones[k[0]]=l_cod
dicc_emisiones[k[1]]=l_nom
dicc_emisiones[k[2]]=l_reg
dicc_emisiones[k[3]]=l_ano
dicc_emisiones[k[4]]=l_co2
dicc_emisiones[k[5]]=l_ccap

#print(dicc_emisiones)

tot_co2=0
n=len(l_cod)
for i in range(0,n):
    if dicc_emisiones[k[2]][i] == 'América Latina y Caribe':
        if dicc_emisiones[k[3]][i] == 2010:
            tot_co2=tot_co2+dicc_emisiones[k[4]][i]

print('El total de emisiones de CO2 para América Latina y Caribe en el año 2010 es: ', tot_co2)


El total de emisiones de CO2 para América Latina y Caribe en el año 2010 es:  1691634.8009999995


2) a) ¿Cuántas variables hay? <br>
        Hay 6 variables: <br>

b) ¿Qué tipos de datos usar para cada una? <br>
                cod_pais = str <br>
                nom_pais = str <br>
                region = str  <br>
                anio = int <br>
                co2 = float <br>
                cor_percapita = float <br>
                
c) ¿Qué tipo de variables son? <br>
                cod_pais = categorica <br>
                nom_pais = categorica <br>
                region = categorica  <br>
                anio =  discreto <br>
                co2 = continuo <br>
                cor_percapita = continuo <br>

d) ¿Hay valores faltantes? <br>
                Si <br>
                
e) ¿Cuál es el total de emisiones de CO2 para 'América Latina y Caribe' en el año 2010? <br>
1691634.8009999995